core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1548,8 +1548,8 @@ class Log(@volatile var dir: File,
         in the header.
       */
       appendInfo.firstOffset match {
-        case Some(firstOffset) => roll(firstOffset)
-        case None => roll(maxOffsetInMessages - Integer.MAX_VALUE)
+        case Some(firstOffset) => roll(Option(firstOffset))
```

Comment:
nit: we can use `Some(firstOffset)` when we have an offset to provide.

Reply:
right, was rushing, will fix.